File: remote/src/main/java/com/kelsos/mbrc/presenters/PlaylistPresenterImpl.java
package com.kelsos.mbrc.presenters;

import com.google.inject.Inject;
import com.kelsos.mbrc.interactors.PlaylistActionInteractor;
import com.kelsos.mbrc.interactors.PlaylistInteractor;
import com.kelsos.mbrc.ui.views.PlaylistListView;
import roboguice.util.Ln;
import rx.android.schedulers.AndroidSchedulers;

public class PlaylistPresenterImpl implements PlaylistPresenter {
  private PlaylistListView view;

  @Inject private PlaylistInteractor playlistInteractor;
  @Inject private PlaylistActionInteractor actionInteractor;

  @Override public void bind(PlaylistListView view) {
    this.view = view;
  }

  @Override public void load() {
    playlistInteractor.execute()
        .observeOn(AndroidSchedulers.mainThread())
        .subscribe(playlists -> view.update(playlists), Ln::v);
  }

  @Override public void play(String path) {
    actionInteractor.play(path).subscribe(aBoolean -> {

    }, throwable -> {

    });
  }
}

File: app/src/main/kotlin/com/kelsos/mbrc/presenters/PlaylistPresenterImpl.kt
package com.kelsos.mbrc.presenters

import com.google.inject.Inject
import com.kelsos.mbrc.interactors.PlaylistActionInteractor
import com.kelsos.mbrc.interactors.PlaylistInteractor
import com.kelsos.mbrc.extensions.task
import com.kelsos.mbrc.ui.views.PlaylistListView
import roboguice.util.Ln

class PlaylistPresenterImpl : PlaylistPresenter {
  private var view: PlaylistListView? = null

  @Inject private lateinit var playlistInteractor: PlaylistInteractor
  @Inject private lateinit var actionInteractor: PlaylistActionInteractor

  override fun bind(view: PlaylistListView) {
    this.view = view
  }

  override fun load() {
    playlistInteractor.allPlaylists.task().subscribe({
      view?.update(it)
    }, { Ln.v(it) })
  }

  override fun play(path: String) {
    actionInteractor.play(path).subscribe({

    }, {

    })
  }
}
